### DISCUSSION TITLE
How to serialize a vector<u8> parameter properly when using Aptos ts-sdk and wallet adapter to submit transaction.

### DISCUSSION THREAD

**Discussion by Metal-joker** [aptos-labs/aptos-developer-discussions#162] [Questions] [answered]
### Describe your question in detail.

I'm trying to call the `0x1::resource_account::create_resource_account_and_publish_package` funciton in ts-sdk and I don't know why I'm getting this error `Move abort 0x10001 in 0x1::utilundefined`
I used plain string I copied from `aptos move build-publish-payload` file for the `metadata_serialized: vector<u8>` arg.
and string array for the `code: vector<vector<u8>>` arg

### What have you tried or looked at?  Or how can we reproduce the error?

```typescript
      const pendingTxn = await signAndSubmitTransaction({
        sender: account.address,
        data: {
          function: "0x1::resource_account::create_resource_account_and_publish_package",
          typeArguments: [],
          functionArguments: [
            "test",
            "0xsomemetadata",
            "0xsomecode",
          ],
        },
      });
```

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment, marked as answer]
This is due to some legacy conversion, so it's converting string to `UTF-8` bytes.  You do need to make sure that they're `Uint8array` or a similar buffer.  To make a `vector<vector<u8>>` you just need to provide an array of `Uint8array`

You can do something like:

```ts
const metadata = "0x12345";
const metadataBytes = Hex.fromHexString(metadata).toUint8Array();

const code = ["0x12345", "0xabcde"];
const codeBytes = code.map((code) => Hex.fromHexInput(code).toUint8Array());

const pendingTxn = await signAndSubmitTransaction({
  sender: account.address,
  data: {
    function: "0x1::resource_account::create_resource_account_and_publish_package",
    typeArguments: [],
    functionArguments: [
      "test",
      metadataBytes,
      codeBytes,
    ],
  },
});

```
